azurelinuxagent/ga/update.py

```diff
@@ -458,6 +460,22 @@ def _initialize_goal_state(self, protocol):
                     logger.info("The current Fabric goal state is older than the most recent FastTrack goal state; will skip it.\nFabric:    {0}\nFastTrack: {1}",
                         egs.created_on_timestamp, last_fast_track_timestamp)
 
+    def _wait_for_cloud_init(self):
+        if conf.get_wait_for_cloud_init() and not self._cloud_init_completed:
+            message = "Waiting for cloud-init to complete..."
+            logger.info(message)
+            add_event(op=WALAEventOperation.CloudInit, message=message)
+            try:
+                output = shellutil.run_command(["cloud-init", "status", "--wait"], timeout=conf.get_wait_for_cloud_init_timeout())
```

Comment:
how long we wait for python2 distros if they set to 'y'?

Reply:
there is no timeout

Reply:
Yes, there is no timeout; it's documented in README. 

The customers requesting this feature are on Python 3. If anybody needs a timeout and is using Python 2, they could implement the timeout on their side, or give us feedback and we may implement it on our side if there is enough feedback. Python 2 has been out if support for some time already.